src/app.rs

```diff
@@ -234,17 +236,16 @@ async fn download_and_cache_s3_object<'a>(
     chunk_cache: &ChunkCache,
     allow_cache_auth_bypass: bool,
 ) -> Result<Bytes, ActiveStorageError> {
-    // We chose a cache key such that any changes to request data
+    // We choose a cache key such that any changes to request data
     // which may feasibly indicate a change to the upstream object
     // lead to a new cache key.
     let key = format!(
-        "{}-{}-{}-{}-{:?}-{:?}",
+        "{}-{}-{}-{:?}-{:?}",
         request_data.source.as_str(),
         request_data.bucket,
         request_data.object,
-        request_data.dtype,
-        request_data.byte_order,
-        request_data.compression,
```

Comment:
My original motivation for including these in the cache key was to try to detect if the object itself had changed (e.g. someone had replaced the named object with the same object but compressed in a different way) but perhaps that's a pathological edge case. Is there much to be gained by removing these from the cache key?

Reply:
As discussed elsewhere, let's just remove these for now since we probably shouldn't be enabling the cache if the underlying data might change. Future feature development task to make cache key configurable may be worthwhile.